### PR TITLE
Add create function for range key

### PR DESCRIPTION
**Pull request by fritzmonkey**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,12 +16,15 @@ class Dynastar {
...
+   *  @param {Function} [options.createHashKey] The function to create the hashKey for certain models (overrides createKey)
+   *  @param {Function} [options.createRangeKey] The function to create the rangeKey for certain models
    */
-  constructor({ model, hashKey = 'key', rangeKey, createKey, ...fns }) {
+  constructor({ model, hashKey = 'key', rangeKey, createKey, createHashKey, createRangeKey, ...fns }) {
     this.model = model;
     this.hashKey = hashKey;
     this.rangeKey = rangeKey;
-    this._createKey = createKey;
+    this._createKey = createHashKey || createKey;
```
Comment (msluther) [inline index.js:26]: Ah, so this is just another name for `createKey`. I assume `createHashKey` is preferred. Are we going to deprecate `createKey` and remove it in the next major version?

File: test/index.test.js [hunk 2]
```diff
@@ -205,6 +205,104 @@ describe('Dynastar - index.js', function () {
...
+          hello: Joi.string(),
+          what: Joi.string(),
+          other: Joi.string().allow(null)
+        }
+      });
+    });
+
+    it('supports createKey for building the hash key', function (done) {
+      const spec = { hello: 'world', what: 'thing' };
+      const myWrapped = new Dynastar({
```
Comment (msluther) [inline test/index.test.js:229]: IMHO, make `myWrapped` be a closure variable (ala model/hash/rangekeys) and then in an `afterEach` drop the tables if `myWrapped` exists.  That way the tests aren't accidentally leaking into each other.

File: index.js (hunk 1 above)
Comment (msluther) [reply index.js:26]: Looks like we never properly documented `createKey` in the `README.md`. Can you add documentation for these two new functions, maybe call out that `createHashKey` will be replacing `createKey`?

**Review by msluther — CHANGES_REQUESTED** [review]
Functionally it looks great. A minor quibble in the tests. But requesting changes here to ensure we properly document the change.

File: test/index.test.js [hunk 3]
```diff
@@ -224,9 +224,15 @@ describe('Dynastar - index.js', function () {
       });
     });
 
+    afterEach(function (done) {
+      if (myWrapped) {
+        myWrapped.dropTables(done);
+      }
+    });
```
Comment (msluther) [inline test/index.test.js:231]:
```suggestion
    afterEach(function (done) {
      if (myWrapped) {
        myWrapped.dropTables(done);
      } else {
        done();
      }
    });
```

File: README.md [hunk 4]
```diff
@@ -63,14 +63,70 @@ const myAwaitModel = new AwaitWrap(mymodel);
...
-// is left untouch but the async callback functon is made to be a `thenable`
+// is left untouched but the async callback function is made to be a `thenable`
 // that can be awaited
 
 const asyncResult = await myAwaitModel.exampleAsyncFn(data);
 const syncResult = myAwaitModel.exampleSyncFn(data);
 
 ```
 
+### key creation
```
Comment (msluther) [inline README.md:74]:
```suggestion
### Key creation
```

**Review by msluther — APPROVED** [review]

File: test/index.test.js [hunk 5]
```diff
@@ -205,6 +205,112 @@ describe('Dynastar - index.js', function () {
...
+      });
+    });
+
+    afterEach(function (done) {
+      if (myWrapped) {
+        myWrapped.dropTables(done);
+      } else {
+        done();
+      }
+    });
```
Comment (msluther) [inline test/index.test.js:233]: Here. this one should pass the linter
```suggestion
      if (myWrapped) {
        return void myWrapped.dropTables(done);
      }
      done();
    });
```

**Review by jcrugzz — APPROVED** [review]

**Pull request merged**